src/Generator/FileGenerator.php

```diff
@@ -336,7 +336,7 @@ public function setClasses(array $classes)
 
     /**
      * @param  string $name
-     * @return ClassGenerator
+     * @return ClassGenerator|bool
```

Comment:
As mentioned in comments below, a type change is a BC break - we should rather identify and fix this edge case scenario, by introducing a fallback or a crash.